core/src/test/java/org/elasticsearch/rest/HeadersAndContextCopyClientTests.java

```diff
@@ -423,4 +421,14 @@ private static void assertContext(ActionRequest<?> request, Map<String, String>
             }
         }
     }
+
+    private static Map<String, String> onlyOnLeft(Map<String, String> left, Map<String, String> right) {
+        Map<String, String> map = new HashMap<>();
+        for (Map.Entry<String, String> entry : left.entrySet()) {
+            if (!right.containsKey(entry.getKey())) {
+                map.put(entry.getKey(), entry.getValue());
+            }
+        }
+        return map;
```

Comment:
@jpountz That's clearer code, and this is just used in tests, but that does have the downside of potentially increasing the size of the array of buckets to be much larger than is needed to hold the difference. That is `{1, 2, ..., n} - {1, 2, ..., n - 1}` would expand as `n` increases but the final size is only one element. The iterative approach maintains the default size. It also allocates a bunch of new `Entry` objects that are ultimately discarded. That's what I was thinking through when I wrote it this way. What do you think?